File: javascript/swea/sw1940/가랏RC카_1940.js
const readline = require('readline');
const fs = require('fs');

const rl = readline.createInterface({
    input: fs.createReadStream('input.txt'),
    output: process.stdout
});

let input = [];
rl.on('line', function (line) {
    input.push(line)
})
    .on('close', function () {
        solution(input);
        process.exit();
    });

function solution(input) {
    let answer = [];

    iter = input[Symbol.iterator]();

    let T = iter.next().value;

    for (let test_case = 1; test_case <= T; test_case++) {
        let speed = 0;
        let distance = 0;

        let N = iter.next().value;

        for (let i = 0; i < N; i++) {
            let line = iter.next().value.split(' ');
            let command = parseInt(line[0]);
            let acceleration = parseInt(line[1]);

            if (command === 1) speed += acceleration;
            else if (command === 2) speed -= acceleration;

            if (speed < 0) speed = 0;

            distance += speed
        }

        answer.push(['#', test_case, ' ', distance, '\n'].join(''))
    }

    console.log(answer.join(''))
}

File: python/swea/sw1940/가랏RC카_1940.py
import sys

def solution():
    answer_list = []

    sys.stdin = open("input.txt", encoding='utf-8', mode='r')

    T = int(input())

    for test_case in range(1, T + 1):
        answer = ['#', str(test_case), ' ', '\n']

        speed = 0
        distance = 0

        N = int(input())
        for i in range(N):
            split_line = input().split()
            if len(split_line) == 1:
                command = split_line
            else:
                command, acceleration = map(int, split_line)

            if command == 1:
                speed += acceleration
            elif command == 2:
                speed -= acceleration

            if speed < 0:
                speed = 0

            distance += speed

        answer.insert(3, str(distance))
        answer_list.append(''.join(answer))

    return ''.join(answer_list)


if __name__ == '__main__':
    print(solution())
